servicetalk-http-netty/src/testFixtures/java/io/servicetalk/http/netty/AsyncContextHttpFilterVerifier.java

```diff
@@ -0,0 +1,205 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.netty;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.concurrent.PublisherSource;
+import io.servicetalk.concurrent.api.AsyncContext;
+import io.servicetalk.concurrent.api.AsyncContextMap;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.concurrent.api.TerminalSignalConsumer;
+import io.servicetalk.http.api.BlockingHttpClient;
+import io.servicetalk.http.api.HttpRequest;
+import io.servicetalk.http.api.HttpServiceContext;
+import io.servicetalk.http.api.StreamingHttpRequest;
+import io.servicetalk.http.api.StreamingHttpResponse;
+import io.servicetalk.http.api.StreamingHttpResponseFactory;
+import io.servicetalk.http.api.StreamingHttpService;
+import io.servicetalk.http.api.StreamingHttpServiceFilter;
+import io.servicetalk.http.api.StreamingHttpServiceFilterFactory;
+import io.servicetalk.http.utils.BeforeFinallyHttpOperator;
+import io.servicetalk.transport.api.HostAndPort;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.InetSocketAddress;
+import java.util.Queue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import static io.servicetalk.concurrent.api.Publisher.from;
+import static io.servicetalk.http.api.HttpSerializationProviders.textDeserializer;
+import static io.servicetalk.http.api.HttpSerializationProviders.textSerializer;
+import static io.servicetalk.http.netty.HttpClients.forSingleAddress;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public final class AsyncContextHttpFilterVerifier {
+
+    private static final AsyncContextMap.Key<String> K1 = AsyncContextMap.Key.newKey("k1");
+    private static final AsyncContextMap.Key<String> K2 = AsyncContextMap.Key.newKey("k2");
+    private static final AsyncContextMap.Key<String> K3 = AsyncContextMap.Key.newKey("k3");
+    private static final AsyncContextMap.Key<String> K4 = AsyncContextMap.Key.newKey("k4");
+
+    private static final String V1 = "v1";
+    private static final String V2 = "v2";
+    private static final String V3 = "v3";
+    private static final String V4 = "v4";
+
+    private AsyncContextHttpFilterVerifier() {
+    }
+
+    public static void verifyServerFilterAsyncContextVisibility(final StreamingHttpServiceFilterFactory filter)
+            throws Exception {
+        final BlockingQueue<Throwable> errors = new LinkedBlockingDeque<>();
+
+        final StreamingHttpServiceFilterFactory filters = asyncContextAssertionFilter(errors).append(filter);
+        final StreamingHttpService service = filters.create(asyncContextRequestHandler(errors));
```

Comment:
Consider adding filters using server builder API instead of manually appending them. Test should reproduce the same behavior what users will do in their apps.